File: hw3/src/main/java/com/yandex/smur/marina/hw3/AddPersonActivity.java
package com.yandex.smur.marina.hw3;

import android.content.Intent;
import android.graphics.Color;
import android.graphics.PorterDuff;
import android.os.Bundle;
import android.os.Message;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;


public class AddPersonActivity extends AppCompatActivity {

    private EditText name;
    private EditText telOrEmail;
    private RadioGroup radioGroup;
    private ImageButton buttonBack;
    private ImageButton buttonSave;

    private int image;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_person);


        name = findViewById(R.id.add_person_name);
        telOrEmail = findViewById(R.id.add_person_phone_email);

        buttonBack = findViewById(R.id.button_back_for_screen_add_person);
        buttonBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });

        buttonSave = findViewById(R.id.button_done_for_screen_add_person);
        buttonSave.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                final Intent intent = new Intent();
                radioGroup = findViewById(R.id.radio_group);
                int checkedId = radioGroup.getCheckedRadioButtonId();
                if (checkedId == -1) {
                } else {
                    findRadioButton(checkedId);
                }

                Contact contact = new Contact(name.getText().toString(),
                        telOrEmail.getText().toString(), image);
                intent.putExtra(Contact.class.getSimpleName(), contact);
                setResult(RESULT_OK, intent);
                finish();
            }

            private void findRadioButton(int checkedId) {
                switch (checkedId) {
                    case R.id.radio_button_phone_number:
                        image = R.drawable.contact_phone;
                        break;
                    case R.id.radio_button_email:
                        image = R.drawable.contact_mail;
                        break;
                    default:
                        break;
                }
            }
        });
    }
}

File: task5/src/main/java/com/yandex/smur/marina/task5/AddPersonActivity.kt
package com.yandex.smur.marina.task5

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_add_person.*


class AddPersonActivity : AppCompatActivity() {

    private var image: Int? = null
    private var id: Double? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_person)


        button_back_for_screen_add_person.setOnClickListener { finish() }

        button_done_for_screen_add_person.setOnClickListener(object : View.OnClickListener {
            override fun onClick(p0: View?) {
                val intent = Intent()
                id = Math.random()
                val checkedId = radio_group.checkedRadioButtonId
                if (checkedId == -1)
                else {
                    findRadioButton(checkedId)
                }
                val contact = Contact(id!!, add_person_name.text.toString(),
                        add_person_info.text.toString(), image!!)
                intent.putExtra(Contact::class.java.simpleName, contact)
                setResult(RESULT_OK, intent)
                finish()
            }

            private fun findRadioButton(checkedId: Int) {
                when (checkedId) {
                    R.id.radio_button_phone_number -> image = R.drawable.contact_phone
                    R.id.radio_button_email -> image = R.drawable.contact_email
                }
            }
        })
    }
}

